File: apps/api_backup_before_import_fix/schemas/project.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, ConfigDict, Field


class ProjectStatus(str, Enum):
    DRAFT = "draft"
    ACTIVE = "active"
    ARCHIVED = "archived"


class ProjectBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=120)
    artist: str = Field(..., min_length=1, max_length=120)
    genre: str = Field(..., min_length=1, max_length=80)
    bpm: int = Field(..., ge=40, le=240)
    key: str = Field(..., min_length=1, max_length=20)
    notes: str = Field(default="", max_length=2000)
    status: ProjectStatus = ProjectStatus.DRAFT


class ProjectCreate(ProjectBase):
    pass


class Project(ProjectBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: apps/web/types/project.ts
export type ProjectStatus = "draft" | "active" | "archived";

export interface Project {
  id: number;
  name: string;
  artist: string;
  genre: string;
  bpm: number;
  key: string;
  notes: string;
  status: ProjectStatus;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreatePayload {
  name: string;
  artist: string;
  genre: string;
  bpm: number;
  key: string;
  notes: string;
  status: ProjectStatus;
}

export interface DashboardSummary {
  projects: number;
  assets: number;
  ai_jobs: number;
  status: string;
}
